server/hotrod/src/main/java/org/infinispan/server/hotrod/ClientListenerRegistry.java

```diff
@@ -158,29 +160,23 @@ void addClientListener(CacheRequestProcessor cacheProcessor, Channel ch, HotRodH
 
       eventSenders.put(new WrappedByteArray(listenerId), clientEventSender);
 
-      if (includeState) {
-         // If state included, do it async
-         CompletableFuture<Void> cf = CompletableFuture.runAsync(() ->
-               addCacheListener(cache, clientEventSender, filter, converter, listenerInterests, useRawData), addListenerExecutor);
-
-         cf.whenComplete((t, cause) -> {
-            if (cause != null) {
-               if (cause instanceof CompletionException) {
-                  cacheProcessor.writeException(h, cause.getCause());
-               } else {
-                  cacheProcessor.writeException(h, cause);
-               }
+      // If state included, do it async
+      CompletionStage<Void> cf = addCacheListener(cache, clientEventSender, filter, converter, listenerInterests, useRawData);
+
+      cf.whenComplete((t, cause) -> {
```

Comment:
`t` to `Void`?

Reply:
Not sure I understand; it is already `Void` through generics. I don't use the argument anyways.

Reply:
I just mean naming the var `Void` to highlight that it's value is never used, but in hindsight that's probably more confusing.

Reply:
I can change it to ignore instead.